### PR TITLE
feat: interactive AskUserQuestion UI for Fresh Claude panes

### PR DESCRIPTION
**Pull request by leegonzales**
## Summary

When Claude calls `AskUserQuestion` in a Fresh Claude pane, the structured question data (options, headers, descriptions) previously rendered as raw JSON inside a `ToolBlock`. This PR adds interactive question UI that users can click to answer, mirroring the existing `PermissionBanner` flow.

- **Server-side interception**: Detects `AskUserQuestion` tool_use blocks in assistant messages, stores pending questions, broadcasts `sdk.question.request` to browser subscribers
- **Client-side rendering**: New `QuestionBanner` component with clickable option buttons, "Other" text input, single/multi-select support, and accessible form controls
- **Answer injection**: User answers are sent back via `sdk.question.respond` WebSocket message, which injects a synthetic `tool_result` into the SDK input stream

## Changes

| Layer | Files | What |
|-------|-------|------|
| Protocol | `shared/ws-protocol.ts`, `server/sdk-bridge-types.ts` | New `SdkQuestionRespondSchema` + `sdk.question.request` message type |
| Server | `server/sdk-bridge.ts`, `server/ws-handler.ts` | Intercept, store, broadcast, respond, inject answer |
| Redux | `src/store/claudeChatTypes.ts`, `src/store/claudeChatSlice.ts` | Question types + `addQuestionRequest`/`removeQuestion` reducers |
| Client | `src/lib/sdk-message-handler.ts` | Handle `sdk.question.request` dispatch |
| UI | `src/components/claude-chat/QuestionBanner.tsx` (new) | Interactive question component |
| Integration | `src/components/claude-chat/ClaudeChatView.tsx` | Wire QuestionBanner into chat view |

## Test plan

- [x] 5 server-side tests (question detection, broadcast, respond, inject, edge cases)
- [x] 7 WS handler tests (schema parsing + routing)
- [x] 2 Redux slice tests (add/remove questions)
- [x] 1 message handler test (dispatches addQuestionRequest)
- [x] 7 QuestionBanner component tests (render, a11y, single-select, multi-select, Other input, submit)
- [x] All 1947 tests pass (`npm test`)
- [x] Type check clean (`tsc --noEmit`)
- [x] Lint clean on new files

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/components/claude-chat/QuestionBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+          return copy
+        })
+      }
+      return { ...prev, [header]: next }
+    })
+  }, [])
+
+  const handleOtherChange = useCallback((header: string, value: string) => {
+    setOtherInputs(prev => ({ ...prev, [header]: value }))
+    setSelections(prev => ({ ...prev, [header]: value }))
```
[the remark this answers is not in the document]
Comment (leegonzales) [reply src/components/claude-chat/QuestionBanner.tsx]: Fixed - handleOtherChange now takes a multiSelect parameter and wraps the value in an array (`[value]`) when true, so multi-select + Other answers are always string[] like regular multi-select responses.

**Comment by justinchoo93** [comment]
Hey @leegonzales — great idea with this PR! The concept of showing an interactive question UI when Claude calls AskUserQuestion is exactly right, and your QuestionBanner component design was solid.

We took your idea and reimplemented it in #112 using the SDK's canUseTool callback instead of intercepting assistant messages. This fixes the blocking issue (Claude now actually waits for the user's answer via a Promise), eliminates the double-prompting (no permission banner alongside the question), and corrects the answer format to match the SDK spec.

Your QuestionBanner UI design carried through — we kept the single-select, multi-select, and "Other" freetext patterns you established.

Closing in favor of #112. Thank you for the contribution!
